src/middlewared/middlewared/plugins/jail.py

```diff
@@ -1377,11 +1453,18 @@ def stop_on_shutdown(self):
     """
     # We need to call a method in Jail service to make sure it runs in the
     # process pool because of py-libzfs thread safety issue with iocage and middlewared
-    if args['id'] == 'ready':
-        try:
-            await middleware.call('jail.start_on_boot')
-        except ioc_exceptions.PoolNotActivated:
-            pass
+    if args['id'] == 'ready' and await middleware.call('jail.iocage_set_up'):
+        if not await middleware.call_sync('system.is_freenas') and await middleware.call_sync(
+            'failover.licensed'
+        ) and await middleware.call('failover.status') == 'MASTER':
+            await middleware.call('jail.check_dataset_existence')
+            await middleware.call('jail.update_defaults', {'nat_backend': 'ipfw'})
+            # We start Jail/VM(s) during carp state change hook
+        else:
+            try:
```

Comment:
We also get here on BACKUP node, is this desired?

Reply:
Nope we don't. Backup node would not return true for `jail.iocage_set_up`

Reply:
Because we only want to use `ipfw` firewall for TrueNAS - so we explicitly check that and update the iocage defaults. Also we don't want to start jails here - we do them in carp state change hook

Reply:
Seems we should always use ipfw for != freenas
And just do the do not start for failover.licensed, no need MASTER check